--- a/CMakeLists.txt	2016-12-16 11:43:00.570163689 +0100
+++ b/CMakeLists.txt	2016-12-16 11:43:07.305215932 +0100
@@ -228,9 +228,9 @@
   SET(${_flags_var_name}      "${${_flags_var_name}} ${Qt5Widgets_EXECUTABLE_COMPILE_FLAGS}")
   SET("CMAKE_CXX_FLAGS_${CMAKE_BUILD_TYPE}" _flags_var_name)
   # prevent error "You must build your code with position independent code if Qt was built with..
-  #if (Qt5_POSITION_INDEPENDENT_CODE)
-  #  SET(CMAKE_POSITION_INDEPENDENT_CODE ON)
-  #endif()
+  if (Qt5_POSITION_INDEPENDENT_CODE)
+    SET(CMAKE_POSITION_INDEPENDENT_CODE ON)
+  endif()
 ELSE()
   SET (QT_MIN_VERSION "4.8.0")
   SET (QT_USE_QTSQL TRUE)
