# HG changeset patch
# User Jan Beich <jbeich@tormail.org>

Bug 783463 - Unbreak --enable-skia linking on Qt X11 targets.

diff --git a/gfx/skia/Makefile.in b/gfx/skia/Makefile.in
index a1a27ca..c995550 100644
--- a/gfx/skia/Makefile.in
+++ b/gfx/skia/Makefile.in
@@ -334,20 +334,19 @@ endif
 ifeq (qt,$(MOZ_WIDGET_TOOLKIT))
 CPPSRCS += \
 	SkFontHost_FreeType.cpp \
 	SkFontHost_gamma_none.cpp \
 	SkFontHost_tables.cpp \
 	SkMMapStream.cpp \
 	SkOSFile.cpp \
 	$(NULL)
-ifeq (Linux,$(OS_TARGET))
+ifneq (,$(MOZ_X11)$(filter Linux,$(OS_TARGET)))
 CPPSRCS += \
 	SkFontHost_linux.cpp \
-	SkFontHost_tables.cpp \
 	SkTime_Unix.cpp \
 	$(NULL)
 endif
 OS_CXXFLAGS += $(MOZ_PANGO_CFLAGS)
 endif
 
 ifeq (windows,$(MOZ_WIDGET_TOOLKIT))
 EXPORTS_skia += \
